server/src/main/java/org/elasticsearch/cluster/metadata/NodesShutdownMetadata.java

```diff
@@ -81,11 +83,42 @@ public void writeTo(StreamOutput out) throws IOException {
     }
 
     /**
-     * Retrieve the data about nodes which are currently in the process of shutting down.
-     * @return A map of node IDs to information about the node's shutdown status.
+     * Retrieve metadata for a single node.
+     * @param nodeId The node ID to get the shutdown metadata for.
+     * @return The shutdown metadata for this node, if it exists, or {@code null} if it does not.
      */
-    public Map<String, SingleNodeShutdownMetadata> getPerNodeInfo() {
-        return Collections.unmodifiableMap(nodes);
+    @Nullable
+    public SingleNodeShutdownMetadata getNodeMetadata(String nodeId) {
+        return nodes.get(nodeId);
+    }
+
+    /**
+     * @return All {@link SingleNodeShutdownMetadata}s that currently exist.
+     */
+    public List<SingleNodeShutdownMetadata> getAllNodeMetadata() {
+        return new ArrayList<>(nodes.values());
```

Comment:
Why not return the `Map` here rather than making a list from the values?

Reply:
I did that at one point, but there's not much additional value in exposing the map I think. If we need to do so, it's easy enough to change.

I don't care strongly about this one way or the other, should I change it?

Reply:
I think I'd prefer the `Map` because I think it'll be a bit easier when testing since we can do get-by-id for nodes

Reply:
Alright, I'll remove the `getNodeMetadata(nodeId)` method above then and replace it with map access as well.